backend/src/main/java/com/capstone/moneytree/controller/UserController.java

```diff
@@ -32,9 +37,59 @@ public UserController(UserService userService) {
         return users;
     }
 
-    @PostMapping("/users")
-    User createUser(@RequestBody User user) {
-        return userService.createUser(user);
+    @PostMapping("/create-user")
+    ResponseEntity<User> createUser(@RequestBody User user) {
+        ResponseEntity<User> response = null;
+        HttpStatus status = validateUserCreation(user);
+
+        if (status == HttpStatus.OK) {
+            User createdUser = userService.createUser(user);
+            response = ResponseEntity.ok(createdUser);
+        }
+
+        return response;
+    }
+
+    /**
+     * A method that updates a user with a new Alpaca key. This should be done only once at beginning
+     * @param userWithKey User with the Alpaca key
+     * @return The new updated user from the database
+     */
+    @PostMapping("/register-alpaca-key")
+    @ModelAttribute
+    ResponseEntity<User> registerAlpacaApiKey(@RequestBody User userWithKey) {
+        String key = userWithKey.getAlpacaApiKey();
+        if (key.isEmpty() || key.isBlank()) {
+            throw new IllegalArgumentException();
+        }
+        User updatedUser = userService.registerAlpacaApiKey(userWithKey);
+
+        if (updatedUser == null) {
+            throw new EntityNotFoundException(ExceptionMessage.ENTITY_NOT_FOUND.getMessage());
+        }
+
+        return ResponseEntity.ok(updatedUser);
+    }
+
+    HttpStatus validateUserCreation(User user) {
+        String email = user.getEmail();
+        String username = user.getUsername();
+        String password = user.getPassword();
+        String firstName = user.getFirstName();
+        String lastName = user.getLastName();
+
+        // TODO: Refactor this into a validation class validateString method to validate strings & message class too
+        if (email.isEmpty() || email.isBlank() ||
```

Comment:
email needs to be unique, we need to add that in the validation!

Reply:
I don't think we need to add this validation as we assume here that the user is already registered (meaning that the email is already unique). That's why I did not add that validation.

Reply:
but we use that method validateUserCreation() in the createUser() endpoint? I have seen that you still make a email validation in the service though so we do not need to add it here anyway. In anycases, Ill pick up the validation user story and refactor so we have a common place for validation.And from there, people can add more validation classes for the new entities what do you think?

Reply:
Yes I agree!